src/utils/flatten.js

```diff
@@ -52,6 +52,11 @@ const flatten = (
       return ruleSet
     }
 
+    /* Throw if a React Element was given styles */
+    if (React.isValidElement(chunk)) {
+      throw new Error('Cannot reference React Elements within styles')
```

Comment:
Let's also beef up this message to make it more useful:

> A React Element (JSX) was interpolated in your styles, probably as a [component selector](https://www.styled-components.com/docs/advanced#referring-to-other-components). Only styled-component classes can be targeted in this fashion.

Reply:
Done.